master/buildbot/test/fake/httpclientservice.py

```diff
@@ -61,6 +61,7 @@ class HTTPClientService(service.SharedService):
 
     def __init__(self, base_url, auth=None, headers=None):
         service.SharedService.__init__(self)
+        assert not base_url.endswith("/"), "baseurl should not end with /"
```

Comment:
I'd rather put it *before* `SharedService.__init__` call: why do extra if things are bad already?